### PR TITLE
eth_getProof added to web3-eth package

### PR DESCRIPTION
**Pull request by nivida**
## Description

This PR does add the JSON-RPC method ``eth_getProof`` to the web3-eth package.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran ```npm run test``` with success and extended the tests if necessary.
- [x] I ran ```npm run build``` and tested the resulting file from ```dist``` folder in a browser.
- [ ] I have tested my code on the live network.

### CODE REVIEW

File: test/eth.getProof.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    formattedResult: formattedProof,
+    call: call
+}, {
+    args: ['0x47d33b27bb249a2dbab4c0612bf9caf4c1950855', [new BigNumber('11')], 'latest'],
+    formattedArgs: ['0x47d33b27bb249a2dbab4c0612bf9caf4c1950855', ['0xb'], 'latest'],
+    result: proof,
+    formattedResult: formattedProof,
+    call: call
+}
+// TODO: Fixing of test.method.js cloning of the arguments does destroy the BN object. Manually calling of utils.numberToHex with new BN('11'); does work.
```
Comment (cgewecke) [inline test/eth.getProof.js:85]: 👍

File: docs/web3-eth.rst [hunk 2]
```diff
@@ -1765,3 +1765,79 @@ Example
...
+
+Returns the account and storage-values of the specified account including the Merkle-proof as described in `EIP-1186 <https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1186.md>`_.
+
+----------
+Parameters
+----------
+
+1. ``String`` 20 Bytes:  The Address of the account or contract.
+2. ``Number[] | BigNumber[] | BN[] | String[]`` 32 Bytes: Array of storage-keys which should be proofed and included. See :ref:`web3.eth.getStorageAt <eth-getStorageAt>`.
+3. ``Number | String | BN | BigNumber``: Integer block number. Pre-defined block numbers as ``"latest"``, ``"earlist"``, ``"pending"``, and ``"genesis"`` can also be used.
```
Comment (cgewecke) [inline docs/web3-eth.rst]:
```suggestion
3. ``Number | String | BN | BigNumber``: Integer block number. Pre-defined block numbers as ``"latest"``, ``"earliest"``, ``"pending"``, and ``"genesis"`` can also be used.
```

In JSON-RPC docs for _getProof it says 
```
QUANTITY|TAG - integer block number, or the string "latest" or "earliest", see the default block parameter
```
Is this discrepancy about whether 'pending' is valid ignorable? I would guess yes but just checking....

**Review by cgewecke — APPROVED** [review]
LGTM. Just left a question about which block params are valid, but looks good!

File: docs/web3-eth.rst (hunk 2 above)
Comment (nivida) [reply docs/web3-eth.rst]: Oh, thanks! I only copied the sentence from above. I will fix the typo here and in all other places (earlist). Thanks!

**Pull request merged**